src/main/java/com/avispl/dal/avdevices/audionetworkinterface/studiotechnologies/danteleaderclock5401a/DanteLeaderClockCommunicator.java

```diff
@@ -217,27 +248,23 @@ private void populateGeneralDTO(Document doc, GeneralDTO generalDTO) {
 	 * Populate statistics for sync input properties
 	 *
 	 * @param stats Map of statistics
+	 * @throws Exception when fail to get sync input statistics
 	 */
-	private void syncInputProperties(Map<String, String> stats) {
-		try {
-			Document doc = Jsoup.parse(doGet(DanteLeaderClockCommands.GET_SYNC_INPUT_COMMAND.getCommand()));
-			SyncInputDTO syncInputDTO = new SyncInputDTO();
-			populateSyncInputDTO(doc, syncInputDTO);
-			if (syncInputDTO.isAllNone()) {
-				throw new ResourceNotReachableException("Fail to populate statistics for SyncInput group");
-			}
-			String groupName = DanteLeaderClockCommands.GET_SYNC_INPUT_COMMAND.getGroupName();
-			stats.put(String.format(DanteLeaderClockConstant.GROUP_PROPERTY_NAME, groupName, DanteLeaderClockMonitoringMetrics.LOCK_STATUS.getPropertyName()), syncInputDTO.getLockStatus());
-			stats.put(String.format(DanteLeaderClockConstant.GROUP_PROPERTY_NAME, groupName, DanteLeaderClockMonitoringMetrics.SYNC_INPUT_STATUS.getPropertyName()), syncInputDTO.getSyncInputStatus());
-			stats.put(String.format(DanteLeaderClockConstant.GROUP_PROPERTY_NAME, groupName, DanteLeaderClockMonitoringMetrics.CURRENT_DANTE_SAMPLE_RATE.getPropertyName()),
-					syncInputDTO.getCurrentDanteSampleRate());
-			stats.put(String.format(DanteLeaderClockConstant.GROUP_PROPERTY_NAME, groupName, DanteLeaderClockMonitoringMetrics.SYNC_INPUT_TYPE.getPropertyName()), syncInputDTO.getSyncInputType());
-			stats.put(String.format(DanteLeaderClockConstant.GROUP_PROPERTY_NAME, groupName, DanteLeaderClockMonitoringMetrics.SYNC_INPUT_TERMINATION.getPropertyName()),
-					syncInputDTO.getSyncInputTermination());
-		} catch (Exception e) {
-			logger.error(e);
-			failedMonitor.put(DanteLeaderClockConstant.SYNC_INPUT, e.getMessage());
+	private void syncInputProperties(Map<String, String> stats) throws Exception {
+		Document doc = Jsoup.parse(doGet(DanteLeaderClockCommands.GET_SYNC_INPUT_COMMAND.getCommand()));
+		SyncInputDTO syncInputDTO = new SyncInputDTO();
+		populateSyncInputDTO(doc, syncInputDTO);
+		if (syncInputDTO.isAllNone()) {
+			throw new ResourceNotReachableException("Fail to populate statistics for SyncInput group");
```

Comment:
also here, is there any reasoning that we could add here, for informational purposes?

Reply:
Let me update this error message as well.